src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -59,6 +61,37 @@ public static int getFileSizeInKB(String filePath) {
         return Math.toIntExact((file.length() / BYTES_PER_KILOBYTES) + 1L); // Add one so that integer division rounds up
     }
 
+    /**
+     * This method retrieves the field mapping by a given field path from the index metadata.
+     *
+     * @param properties Index metadata mapping properties.
+     * @param fieldPath The field path string that make up the path to the field mapping. e.g. "a.b.field" or "field".
+     *                  The field path is applied and checked in OpenSearch, so it is guaranteed to be valid.
+     *
+     * @return           The field mapping object if found, or null if the field is not found in the index metadata.
+     */
+    private static Object getFieldMapping(final Map<String, Object> properties, final String fieldPath) {
+        String[] fieldPaths = fieldPath.split("\\.");
+        Object currentFieldMapping = properties;
+
+        // Iterate through the field path list to retrieve the field mapping.
+        for (String path : fieldPaths) {
+            currentFieldMapping = ((Map<String, Object>) currentFieldMapping).get(path);
+            if (currentFieldMapping == null) {
+                return null;
+            }
+
+            if (currentFieldMapping instanceof Map<?, ?>) {
```

Comment:
If we are casting to Map<String, Object> in line 85, can this check be Map<String, Object>?